File: TradingJournal.UI/src/app/interfaces/rest/stock-data-dto.ts
import { SymbolDto } from "./symbol-dto";
import { HistoricalBarDto } from "./historical-bar-dto";

export interface StockDataDto {
    ticker?: string;
    companyName?: string;
    price?: {
        value: number,
        currency: string,
        lastUpdated: Date
    }
    change?: number;
    changePercent?: number;
    volume?: number;
    avgVolume?: number;
    marketCap?: number;
    peRatio?: number;
    week52High?: number;
    week52Low?: number;
    ytdChange?: number;
    lastUpdated?: string;
    stockLogo?: {
        name: string,
        ticker: string,
        image: string,
    };
    historicalBars?: HistoricalBarDto[];
}

File: TradingJournal.API.dotnet/TradingJournal.API/Dtos/StockDataDto.cs
using TradingJournal.Models.Api_Ninja;

namespace TradingJournal.Dtos;

public class StockDataDto
{
    public string Ticker { get; set; } = string.Empty;
    public string CompanyName { get; set; } = string.Empty;
    public Price? Price { get; set; }
    public double Change { get; set; }
    public double ChangePercent { get; set; }
    public double Volume { get; set; }
    public double AvgVolume { get; set; }
    public double MarketCap { get; set; }
    public double PeRatio { get; set; }
    public double Week52High { get; set; }
    public double Week52Low { get; set; }
    public double YtdChange { get; set; }
    public string LastUpdated { get; set; } = string.Empty;
    public Logo? StockLogo { get; set; }
    public HistoricalBarDto[] HistoricalBars { get; set; } = [];
}

public class Price
{
    public double Value { get; set; }
    public string Currency { get; set; } = string.Empty;
    public DateTime LastUpdated { get; set; }
}